### PR TITLE
Feature/Modify code quality tools

### PR DESCRIPTION
**Pull request by SnowSuno**
This PR adds in `pretter` and changes some existing rules of `eslint`. There are also dependency version changes of `prettier` and `eslint`, and its dependencies.

- misc(prettier): Add prettier settings
- misc(eslint): Seperate eslint settings to `.eslintrc.json`
- misc(eslint): Update old dependencies of eslint related modules
- misc(eslint): Downgrade `eslint` to 8.22.0
- misc(prettier): Configure prettier
- misc(prettier): Add prettier hooks
- misc(prettier): Change prettier rules
- style(prettier): Apply prettier styles

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -84,44 +87,5 @@
         }
       ]
     ]
-  },
-  "eslintConfig": {
```
Comment (withSang) [inline package.json:88]: 👍

File: src/utils/sso.js [hunk 2]
```diff
@@ -1,5 +1,9 @@
-import Client from './sparcsssov2-node';
+import Client from "./sparcsssov2-node";
 
-const client = new Client (process.env.SSO_CLIENT_ID, process.env.SSO_SECRET, false); // TODO: MOVE TO DOTENV
+const client = new Client(
+  process.env.SSO_CLIENT_ID,
+  process.env.SSO_SECRET,
+  false,
+); // TODO: MOVE TO DOTENV
```
Comment (withSang) [inline src/utils/sso.js:7]: TODO: remove TODO

File: src/utils/logger/signale.js [hunk 3]
```diff
@@ -1,29 +1,29 @@
...
+      color: "yellow",
+      label: "reminder",
+      logLevel: "info",
     },
     santa: {
-      badge: '🎅',
-      color: 'red',
-      label: 'santa',
-      logLevel: 'info',
+      badge: "🎅",
```
Comment (withSang) [inline src/utils/logger/signale.js:12]: 와우 이런 로깅 라이브러리도 썼군요

**Review by withSang — APPROVED** [review]
Looks prettier 😲

File: src/utils/logger/signale.js (hunk 3 above)
Comment (SnowSuno) [reply src/utils/logger/signale.js:12]: 아니 이거 뭐야 ㅋㅋㅋㅋ

**Pull request merged**